### PR TITLE
Global trustDomain setting is deprecated in favour of meshConfig.trustDomain

### PR DESCRIPTION
**Pull request by shamsher31**
Part of https://github.com/istio/istio/issues/27734

Related: https://github.com/istio/istio/pull/27806

### CODE REVIEW

**Comment by frankbu** [comment]
@shamsher31 It seems that this change causes the test to fail. I thought it might have just been a flake, but it seems to have failed again after a retest.

**Comment by shamsher31** [comment]
@frankbu let me check.

**Comment by howardjohn** [comment]
I will send a fix for this.

**Review by howardjohn — COMMENTED** [review]
You need to change `Install Istio with a new trust domain and trust domain aliases.` as well

I am not actually sure how that one works - I am nearly certain its broken in 1.7 code? is it skipped in the test?

File: content/en/docs/tasks/security/authorization/authz-td-migration/index.md [hunk 1]
```diff
@@ -83,10 +83,10 @@ Notice that it may take tens of seconds for the authorization policy to be propa
 
 ## Migrate trust domain without trust domain aliases
 
-1. Install Istio with a new trust domain.
+1. Install Istio with a new trust domain and trust domain aliases.
```
Comment (frankbu) [inline content/en/docs/tasks/security/authorization/authz-td-migration/index.md:86]: This is wrong. This section does NOT use a trust domain alias (see line 84).
```suggestion
1. Install Istio with a new trust domain.
```

Comment (frankbu) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md:86]: @howardjohn's comment was referring to the example on line 132. That needs to be changed.

Comment (shamsher31) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md:86]: Sorry, will update.

Comment (shamsher31) [reply content/en/docs/tasks/security/authorization/authz-td-migration/index.md]: Done

**Comment by howardjohn** [comment]
master fix is merged, https://github.com/istio/istio/pull/28293 is on 1.8

**Comment by frankbu** [comment]
Done in #8399

**Pull request closed without merging**